From 9cfccdb23572c08ab3d051e4a4262c1a78affae2 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 17 Jun 2021 15:30:35 +0300
Subject: [PATCH 04/14] s32: Don't enable caches for secondary cores

Enabled caches for secondaries can corrupt Linux environment when
the lines are flushed. The corruption of Linux user space is also
expected.

Upstream-Status: Pending

Issue: ALB-6839
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/lowlevel.S | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/arch/arm/cpu/armv8/s32/lowlevel.S b/arch/arm/cpu/armv8/s32/lowlevel.S
index bebb5548c5..75af957762 100644
--- a/arch/arm/cpu/armv8/s32/lowlevel.S
+++ b/arch/arm/cpu/armv8/s32/lowlevel.S
@@ -123,13 +123,6 @@ start_slave_cores:
 	msr	mair_el3, x0
 	msr	mair_el2, x0
 
-	/* icache & dcache */
-	mrs	x0, sctlr_el3
-	ldr	x1, =(1<<12) | (1<<5) | (1 << 3) | (1<<2) |(1 << 0) /* bits SA(3)  M(0) */
-	orr	x0, x0, x1     /* set bits */
-	msr	sctlr_el3, x0
-	isb
-
 	/*
 	 * MPIDR_EL1 Fields:
 	 * MPIDR[1:0] = AFF0_CPUID <- Core ID (0,1)
-- 
2.17.1

